src/dune_lang/action.ml

```diff
@@ -398,6 +398,31 @@ let rec map_string_with_vars t ~f =
   | Pipe (o, ts) -> Pipe (o, List.map ts ~f:(map_string_with_vars ~f))
   | Cram sw -> Cram (f sw)
 
+let rec list_of_string_with_vars t =
```

Comment:
looks like there's a bunch of unrelated code that has been pulled in?

Reply:
Recycling the old tests, I find out there was `lib-available:<library-name>` and it's considered like optional dependency.

Reply:
When you cherry-picked old tests in a8b62720b907d23a8373ea30b261cbded42ed318, the commit also added some changes in `src/`. Can you remove those?

Reply:
I didn't a cherry-pick, the tests was not consequent. So I just did a copy paste mentioning it was the all tests.

Reply:
OK I understand now. I don't think that this approach is going to work, since it only collects pforms in `(rule)`. Can we stick to the simpler version (that does not try to collect lib names in pforms) for 3.5? Otherwise we can try something that somehow folds over pforms in all stanzas, but this is going to be more involved so we can't do that before 3.6.